processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -266,9 +267,15 @@ public void initiateShutdown() {
     }
 
     @Override
-    public void awaitShutdown() throws InterruptedException {
-        join();
+    public boolean awaitShutdown(Duration limit) throws InterruptedException {
+        join(limit.toMillis());
         metrics.close();
-        log.info("Subscription thread terminated: {}", getName());
+        if (isAlive()) {
+            log.warn("Subscription thread didn't terminate within time limit: {}", getName());
```

Comment:
Happy to lower it to info, but I think we do want to know which case applied and it's not always clear from outside (we return a boolean but that doesn't necessarily tell us the details of which subpartition didn't terminate).

Reply:
> which subpartition

which subscription you mean?

Can we check "absence" of the below log instead?
 https://github.com/line/decaton/blob/e0b209d89922cc98b9568f6d8b08e37f5294ed7b/processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java#L256

Anyway, my opinion for such logging is that its allowed only when:
* The interface intends to complete the whole shutdown process within the method call (either with or without timeout). I think `awaitShutdown` doesn't intend that (it observers shutdown for the given amount of time and just returns the result)  or,
* termination failed with obvious error (e.g, exception)

Otherwise, we're putting assumption on it such that this is the single last invocation from user and they'll leak some resources if we don't complete everything here, such might not be a case, making it inconsistent with interface design I think.

Still fine to have such logging but debug level is appropriate I think? (if the above "absence" check works)